docs/content/v0-upgrade.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Upgrading to v1.0
+kind: documentation
+weight: 122
+---
+
+All users should plan to upgrade to OPA v1.0 eventually. Some users, with more
+control over the Rego loaded into the OPA instances they run will be able to do
+so more quickly. Other users with less control or running third party Rego may
+wish to upgrade to OPA v1.0 and use the v0 compatibility functionality to
+upgrade gradually.
+
+This documentation covers the different upgrade scenarios and the best course of
+action for each. The documentation makes use of the following concepts:
+
+- **Bundle Producer**: A system based on `opa build` that produces a bundle that
+  is loaded by consumers.
+- **Bundle Consumer**: An OPA instance running that loads and evaluates policy
+  from a bundle in the system.
+- **Authoring**: The process of writing Rego policies before bundles are
+  produced and consumed. In managed systems, this might be a user's only contact
+  point with OPA.
+
+In some systems, where OPA is used without a bundle, there is no producer. This
+simplifies the upgrade process.
+
+## General Upgrade Approach: Upgrade Producers, then Consumers
+
+Users will need to upgrade to OPA v1.0 in their own way, depending on their
+release and change management processes, use cases and risk tolerance. The
+general advice is to upgrade producers first, then consumers. Some users may
```

Comment:
We should add that the reason for why we recommend upgrading producers first is because they'll set a rego-version flag in compiled bundles, which means that consumers can be run without `--v0-compatible` flags (launch command can be kept agnostic). Also, it's more likely that you have several consumers but just a few producers (singular, even).

Reply:
Updated